//-----------------------------------------------------------------------------
//
//  Copyright (C) 2010-2012 Artem Rodygin
//
//  This file is part of EncMQ.
//
//  EncMQ is free software: you can redistribute it and/or modify
//  it under the terms of the GNU General Public License as published by
//  the Free Software Foundation, either version 3 of the License, or
//  (at your option) any later version.
//
//  EncMQ is distributed in the hope that it will be useful,
//  but WITHOUT ANY WARRANTY; without even the implied warranty of
//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//  GNU General Public License for more details.
//
//  You should have received a copy of the GNU General Public License
//  along with EncMQ.  If not, see <http://www.gnu.org/licenses/>.
//
//-----------------------------------------------------------------------------

package message;

//-----------------------------------------------------------------------------
//  General envelope for plain messages.
//-----------------------------------------------------------------------------

message envelope
{
    // message body
    required string msg_type    = 1;    // message type (descriptor)
    required uint32 msg_size    = 2;    // size of message body (in bytes)
    optional bytes  msg_data    = 3;    // message body

    // message authentication code
    optional bytes  mac         = 4;

    // message extensions
    extensions 100 to max;
};

//-----------------------------------------------------------------------------
//  Secured envelope for encrypted messages.
//-----------------------------------------------------------------------------

extend envelope
{
    optional bytes  key         = 100;  // AES symmetric key
};
